swath_projector/utilities.py

```diff
@@ -216,3 +225,24 @@ def make_array_two_dimensional(one_dimensional_array: np.ndarray) -> np.ndarray:
 
     """
     return np.expand_dims(one_dimensional_array, 1)
+
+
+def transpose_if_xdim_less_than_ydim(variable: np.ndarray) -> np.ndarray:
+    """Return transposed variable when variable is wider than tall.
+
+    QuickFix (DAS-2216): We presume that a swath has more rows than columns and
+    if that's not the case we transpose it so that it does.
+    """
+    if variable.ndim != 2:
+        raise ValueError(
+            f'Input variable must be 2 dimensional, but got {variable.ndim} dimensions.'
+        )
+    if variable.shape[0] < variable.shape[1]:
+        transposed_variable = np.ma.transpose(variable).copy()
+        if hasattr(variable, 'mask'):
```

Comment:
Do we have datasets with a "mask" attribute, or is this something that is located somewhere else or added later?

Reply:
Yes - all datasets come through to this point as Numpy Masked-Array (np.ma) datasets.  It is, behind the scenes, the default behavior of the NetCDF library.  We have to acknowledge that in terms of fill data being 'masked' in the dataset arrays, not fill values.  The code works with np.ma functions in a number of cases, and has to in the case of transpose in order not to lose the data mask component of Masked-Array datasets.